### ISSUE TITLE
Correct Modularisation of Application to adhere to StandardJS

### ISSUE THREAD

**Issue by quadrophobiac** [theodi/comma-chameleon#180]
[PLACEHOLDER, will be edited more comprehensively, relates to https://github.com/theodi/comma-chameleon/pull/177]
> Please provide a general summary of the issue in the Issue Title above
> fill out the headings below as applicable to the issue you are reporting,
> deleting as appropriate but offering us as much detail as you can to help us resolve the issue

### Expected Behaviour
> What should happen?

### Desired Behaviour (for enhancement suggestions only)
> if relevant include images or hyperlinks to other resources that clarify the enhancement you're seeking

### Current Behaviour (for problems)
> What currently happens that isn't expected behaviour?

### Steps to Reproduce (for problems)
> Provide a link to a live example, or an unambiguous set of steps to reproduce this bug. Include code to reproduce, if relevant
1.
2.
3.
4.

### Your Environment
> Include as many relevant details about the environment you experienced the bug in - this will help us resolve the bug more expediently
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):

**Comment by chris48s** [comment]
Are those the remaining errors you've got left from running the standard js checks after the commits you've already made in #177 ?

It is kind of difficult to generalise about all of this as it seems like the warnings you've posted there are caused by a variety of different issues. For example:

* If you look at `'worksheet' is assigned a value but never used` in `excel.js`, that is just a simple unused variable - its assigned in the function scope and not used. You can resolve that by just deleting line 9 (and then 8 as this is only used by 9). The same is true for `'window' is assigned a value but never used` in `file.js`. They're just cruft that can be deleted.

* I think all the uses of `mainWindow` as a global have now been removed, so you should be able to just delete `var mainWindow = null` from `main.js` and nothing will actually change.

* With `menu` in `main.js`, it seems you can just make that line `var menu = Menu.buildFromTemplate(template);` and bring it into the local scope without changing anything. I'm not really sure why `jshint` wasn't warning about this missing `var` statement :/ The call to `menu.popup()` in `index.js` seems to be using the value of `menu` defined in `renderer/menu.js` - not the value that is being defined in `main.js`. Note that if you change that line to `Menu.getApplicationMenu().popup(remote.getCurrentWindow());` it does something completely different. In order to remove that `menu` instance from the global scope, I think you probably want to refactor `renderer/menu.js` into a module which exports `menu` and then `require` it in `index.js` but I've not checked that.

* The issues with `githubWindow` and `parentWindow` will need some refactoring to properly sort out. Basically, `authAndLoad()` is polluting the global scope with those 2 variables and then all the other functions can modify them but I think those should be defined in the local scope of `authAndLoad()` and all your functions that referece `githubWindow` or `parentWindow` should accept the window as a parameter (in the same way that `loadWindow()` and `uploadToGithub()` do) and those vars should be passed around.. or maybe you could turn it into a module with `githubWindow` and `parentWindow` as private properties? Again I'm not sure why `jshint` wasn't throwing a wobbly over this one. Personally I would tend to tackle this in 2 parts:

    * Leave `parentWindow` and `githubWindow` in the global scope but make it explicit they are global by using the `global.` prefix and concentrate on conforming to the coding standards for now (i.e: the scope of PR #177 should just be to maintain the existing functionality but comply with the new coding standards)
    * Once that is merged, start a new issue/PR to refactor `github.js`, removing `parentWindow` and `githubWindow` from the global scope

In general, it is a good idea to remove vars from the global scope and one of the handy things that PR #177 is helping with is identifying the size of that problem and making it explicit but I think its better to avoid doing too much at once if major refactoring is needed.

I've not been through every single warning, but hopefully some of the rest of them will follow the same patterns I've identified there. See how far you get with those suggestions and give me a shout if there's anything else you are stuck on.
